src/PythonQtPythonInclude.h

```diff
@@ -142,4 +142,19 @@
 #define PyBytes_FromStringAndSize PyString_FromStringAndSize
 #endif
 
+// Avoid clashes with libstdc++ <locale> by undefining ctype macros
+// that CPython may introduce on macOS when the UTF-8 ctype quirk is enabled.
+// (_PY_PORT_CTYPE_UTF8_ISSUE is defined by CPython’s pyport.h; we apply these
+// undefs only in C++ builds.)
```

Comment:
Interesting. Searching for this reveals a number of similar issues in CPython itself, which they seem to have resolved by using the macros from `pyctype.h` like `Py_ISALPHA` etc.